controllers/ext_client.go

```diff
@@ -248,9 +249,9 @@ func createExtClient(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 
 	var params = mux.Vars(r)
-
 	networkName := params["network"]
 	nodeid := params["nodeid"]
+	clientid := params["clientid"]
```

Comment:
client id should be retrieved from struct passed in request.Body not from url parameter

Reply:
OK but i need to use the ExtClient struct or create a new struct for this type of request?

Reply:
you could do either ... use the ExtClient struct but only send the ClientID param in the body of the request or use a new struct.
I would create an anonymous struct consisting of just one element and decode the body into that struct and then copy to extclient struct.  
eg.

request := struct {
     id string
}
if err := json.NewDecoder(r.body).Decode(&request); err != nil {
   //log error
} else {
   extclient.ClientID = request.id
}

Reply:
I have made the changes